src/plugins/graph/models/graph-model.ts

```diff
@@ -74,14 +76,43 @@ export const GraphModel = TileContentModel
     showMeasuresForSelection: false
   })
   .volatile(self => ({
-    prevDataSetId: "",
-    autoAssignedAttributes: [] as Array<{ place: GraphPlace, role: GraphAttrRole, dataSetID: string, attrID: string }>,
-    disposeDataSetListener: undefined as (() => void) | undefined
+    // prevDataSetId: "",
+  }))
+  .preProcessSnapshot((snapshot: any) => {
+    const hasLayerAlready:boolean = (snapshot?.layers?.length || 0) > 0;
+    if (!hasLayerAlready && snapshot?.config) {
+      const layer = { config: snapshot.config };
+      snapshot.layers = [layer];
+      delete(snapshot.config);
```

Comment:
`preProcessSnapshot` is supposed to return a new snapshot rather than modifying the one passed in. For instance, something like:
```
  // remove config from snapshot
  const { config, ...others } = snapshot
  return config != null ? { ...others } : snapshot
```